staging/src/kubevirt.io/api/apitesting/testdata/HEAD/kubevirt.io.v1.KubeVirt.json

```diff
@@ -152,7 +152,8 @@
               },
               "claims": [
                 {
-                  "name": "nameValue"
+                  "name": "nameValue",
+                  "request": "requestValue"
```

Comment:
What is this?

Reply:
Looks like they comes from `./hack/update-generated-api-testdata.sh` executed during `make generate`.
Don't know more. @Sreeja1725 can you help here?

Reply:
It seems by updating the dependencies we expand our API. The question is do we want to?

Reply:
This comes from the new ResourceRequirements field: https://kubernetes.io/docs/reference/kubernetes-api/workload-resources/pod-v1/#resources:

>ResourceClaim references one entry in PodSpec.ResourceClaims.
>- resources.claims.name (string), required
Name must match the name of one entry in pod.spec.resourceClaims of the Pod where this field is used. It makes that resource available inside a container.
>- resources.claims.request (string)
Request is the name chosen for a request in the referenced claim. If empty, everything from the claim is made available, otherwise only the result of this request.

And since the ResourceRequirements are used by `ComputeResourceOverhead` it results in the update.
FWIW It is an optional field

Reply:
This is an excellent example of this tool doing its job. cc @EdDev 

Both the places (KubevirtSpec and VirtualMachineSpec) where ResourceRequirements changes were caught by the tool, it does not make sense for DRA devices to be consumed.  Even if they could be used to consume DRA devices, it will have to be after DRA devices are supported in KubeVirt.

Therefore I recommend to change both the api structs as follows:

```
type InterfaceBindingPlugin struct {
...
	// ComputeResourceOverhead specifies the resource overhead that should be added to the compute container when using the binding.
	// version: v1alphav1
	// +optional
	ComputeResourceOverhead *ResourceRequirementsWithoutClaims `json:"computeResourceOverhead,omitempty"`
}

type ResourceRequirementsWithoutClaims struct {
	// Limits describes the maximum amount of compute resources allowed.
	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
	// +optional
	Limits ResourceList `json:"limits,omitempty" protobuf:"bytes,1,rep,name=limits,casttype=ResourceList,castkey=ResourceName"`
	// Requests describes the minimum amount of compute resources required.
	// If Requests is omitted for a container, it defaults to Limits if that is explicitly specified,
	// otherwise to an implementation-defined value. Requests cannot exceed Limits.
	// More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/
	// +optional
	Requests ResourceList `json:"requests,omitempty" protobuf:"bytes,2,rep,name=requests,casttype=ResourceList,castkey=ResourceName"`
}
```

Reply:
> And since the ResourceRequirements are used by ComputeResourceOverhead it results in the update.
FWIW It is an optional field

While I understand it is optional field, removing a field once introduced into the API has proven to be extremely challenging. If the fields are being added just because of k8s deps bump without any usecase, we should try to not extend the api exposure.

Reply:
Fair enough from my side. This will also generate a change from the tool, since the claims are not present anymore, but in this case it is acceptable, right?

Reply:
Fair enough from my side. This will also generate a change from the tool, since the claims are not present anymore, but in this case it is acceptable, right?